atlasdb-client/src/main/java/com/palantir/atlasdb/tracing/CloseableTrace.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.tracing;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.helpers.MessageFormatter;
+
+import com.google.common.collect.ObjectArrays;
+import com.palantir.remoting1.tracing.OpenSpan;
+import com.palantir.remoting1.tracing.SpanType;
+import com.palantir.remoting1.tracing.Tracer;
+
+public final class CloseableTrace implements AutoCloseable {
+
+    private static final CloseableTrace NO_OP = new CloseableTrace(null);
+
+    private final OpenSpan trace;
+
+    private CloseableTrace(@Nullable OpenSpan trace) {
+        this.trace = trace;
+    }
+
+    @Override
+    public void close() {
+        if (trace != null) {
+            Tracer.completeSpan();
+        }
+    }
+
+    public static CloseableTrace startLocalTrace(String serviceName,
+            CharSequence operationFormat,
+            Object[] formatArguments) {
+        if (Tracer.isTraceObservable()) {
+            return CloseableTrace.startLocalTrace("{}." + operationFormat,
+                    ObjectArrays.concat(serviceName, formatArguments));
+        }
+        return NO_OP;
+    }
+
+    private static CloseableTrace startLocalTrace(CharSequence operationFormat, Object... formatArguments) {
+        if (Tracer.isTraceObservable()) {
```

Comment:
given this is private this protection seems unnecessary.  Maybe a Precondition check instead